runtime/common/src/precompile/honzon.rs

```diff
@@ -0,0 +1,523 @@
+
+		match action {
+			Action::AdjustLoan => {
+				let who = input.account_id_at(1)?;
+				let currency_id = input.currency_id_at(2)?;
+				let collateral_adjustment = input.i128_at(3)?;
+				let debit_adjustment = input.i128_at(4)?;
+
+				log::debug!(
+					target: "evm",
+					"honzon: adjust_loan who: {:?}, currency_id: {:?}, collateral_adjustment: {:?}, debit_adjustment: {:?}",
+					who, currency_id, collateral_adjustment, debit_adjustment
+				);
+
+				<module_honzon::Pallet<Runtime> as HonzonManager<Runtime::AccountId, CurrencyId, Amount, Balance>>::adjust_loan(&who, currency_id, collateral_adjustment, debit_adjustment).map_err(|e|
+                    PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: Into::<&str>::into(e).as_bytes().to_vec(),
+                        cost: target_gas_limit(target_gas).unwrap_or_default(),
+                    }
+                )?;
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: vec![],
+					logs: Default::default(),
+				})
+			}
+			Action::CloseLoanByDex => {
+				let who = input.account_id_at(1)?;
+				let currency_id = input.currency_id_at(2)?;
+				let max_collateral_amount = input.balance_at(3)?;
+
+				log::debug!(
+					target: "evm",
+					"honzon: close_loan_by_dex who: {:?}, currency_id: {:?}, max_collateral_adjustment: {:?}",
+					who, currency_id, max_collateral_amount
+				);
+
+				<module_honzon::Pallet<Runtime> as HonzonManager<Runtime::AccountId, CurrencyId, Amount, Balance>>::close_loan_by_dex(who, currency_id, max_collateral_amount).map_err(|e|
+                    PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: Into::<&str>::into(e).as_bytes().to_vec(),
+                        cost: target_gas_limit(target_gas).unwrap_or_default(),
+                    }
+                )?;
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: vec![],
+					logs: Default::default(),
+				})
+			}
+			Action::GetPosition => {
+				let who = input.account_id_at(1)?;
+				let currency_id = input.currency_id_at(2)?;
+
+				let Position { collateral, debit } = <module_honzon::Pallet<Runtime> as HonzonManager<
+					Runtime::AccountId,
+					CurrencyId,
+					Amount,
+					Balance,
+				>>::get_position(&who, currency_id);
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128_tuple(collateral, debit),
+					logs: Default::default(),
+				})
+			}
+			Action::GetLiquidationRatio => {
+				let currency_id = input.currency_id_at(1)?;
+				let ratio = <module_honzon::Pallet<Runtime> as HonzonManager<
+					Runtime::AccountId,
+					CurrencyId,
+					Amount,
+					Balance,
+				>>::get_liquidation_ratio(currency_id)
+				.ok_or(PrecompileFailure::Revert {
```

Comment:
we should not revert here
```suggestion
				.unwrap_or_default()
```